ipc_receivefile.c

```diff
@@ -0,0 +1,400 @@
+////////////////////////////////////////////////////////////////////////////////
+//IPC RECEIVER
+//
+//
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <sys/iofunc.h>
+#include <sys/dispatch.h>
+#include <sys/neutrino.h>
+
+#include "ipc_common.h"
+
+#include <getopt.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <sys/mman.h>
+#include <process.h>
+#include <string.h>
+#include <unistd.h>
+
+typedef union
+{
+	uint16_t msg_type;
+	struct _pulse pulse;
+	cksum_header_t cksum_hdr;
+} msg_buf_t;
+
+typedef union
+{
+	uint16_t type;
+	struct _pulse pulse;
+	get_shmem_msg_t get_shmem;
+	changed_shmem_msg_t changed_shmem;
+} recv_buf_t;
+
+void ipcMessagePassingReceive(int fd);
+int create_shared_memory(unsigned nbytes, int client_pid, void **ptr, shm_handle_t *handle);
+void ipcSharedMemoryReceive(int fd);
+void ipcPipeReceive(int fd);
+
+int main(int argc, char* argv[])
+{
+    int fd = -1;
+    int longindex;
+    int ch;
+    char method;
+    static struct option longopts[] = {
+        { "help", no_argument,  NULL, 'h' },
+		{ "messages", no_argument,  NULL, 'm' },
+        { "queue", no_argument,  NULL, 'q' },
+        { "pipe",   no_argument, NULL, 'p' },
+        { "shm",   no_argument, NULL, 's' },
+        { "file",   required_argument, NULL, 'f' },
+        { NULL, 0, NULL, 0 }
+    };
+
+    while ((ch = getopt_long(argc, argv, "hmqpsf:", longopts, &longindex)) != -1) {
+
+        switch (ch) {
+        case 'f':
+                if ((fd = open(optarg, O_WRONLY)) == -1)

```

Comment:
O_CREAT as flag or we can't create the file.

Reply:
A file should be created before running.

Reply:
It could be just easier for you and the user to add the flag O_CREAT...